---
title: "FMP functions analysis"
output: html_notebook
---

```{r message=FALSE, warning=FALSE}

library(ggplot2)
library(dplyr)
devtools::load_all()


```




```{r}

data("kegg")
data("cazy")
data("MSP_tax")
data("sample_metadata_metaG")

```


```{r}

load("../data-raw/tolerance_MSP_table.rda")

counts_scaled = vroom::vroom("../data-raw/scaled_normalization_counts.txt")

table(kegg$KO) %>% dim



```




```{r}


sample_metadata_metaG = data.frame(seq=colnames(counts_scaled)[-1], samplenr=colnames(counts_scaled)[-1] %>% stringr::str_split_fixed(pattern="_", 2) %>% .[,1]) %>%
  merge(sample_metadata_metaG, by="samplenr")


```




```{r}
head(cazy)
head(kegg)
head(tolerance_MSP_table)
head(MSP_tax)
head(counts_scaled)

```


## check gene labelling

```{r}


counts_scaled %>%
  sample_frac(0.01) %>%
  select(1)

tolerance_MSP_table %>% 
  sample_frac(0.01) %>% 
  select(gene_name) %>%
  filter(gene_name %in% grep("Lr10",gene_name, value=TRUE ))

# to remove "|MetaHIT" everywhere

```


```{r eval=FALSE, include=FALSE}

counts_scaled %>%  #sample_frac(0.01) %>%
  select(1) %>%
  rename(gene_name = 1) %>%
  merge(kegg, by="gene_name", all.x=TRUE) %>%
  merge(cazy, by="gene_name", all.x=TRUE) %>%
  mutate(gene_name = gene_name %>% gsub("\\|MetaHIT","",.)) %>%
  merge(tolerance_MSP_table,by="gene_name", all.x=TRUE) -> gene_description



```

```{r}

cazy <- cazy %>% mutate(gene_name = gene_name  %>% gsub("\\|MetaHIT","",.)) %>% mutate(gene_name = gene_name %>% gsub("\\|[0-9]*_aa","",.))
kegg <- kegg %>% mutate(gene_name = gene_name  %>% gsub("\\|MetaHIT","",.)) %>% mutate(gene_name = gene_name %>% gsub("\\|[0-9]*_aa","",.))
counts_scaled <- counts_scaled %>% rename(gene_name = 1) %>% mutate(gene_name = gene_name  %>% gsub("\\|MetaHIT","",.)) %>% mutate(gene_name = gene_name %>% gsub("\\|[0-9]*_nt","",.))
tolerance_MSP_table <- tolerance_MSP_table %>% mutate(gene_name = gene_name %>% gsub("\\|[0-9]*_nt","",.))


```



```{r eval=FALSE, include=FALSE}

cazy %>%
  tidyr::separate(gene_name,  into=c("gene_id","project"), sep="\\|")





```



```{r eval=FALSE, include=FALSE}
#msp_lrhamnosus
#msp_1025

tolerance_MSP_table %>% head


rbind(

kegg %>%
  filter(gene_name %in% grep("Lr10",gene_name, value = TRUE)) %>%
  tidyr::separate(gene_name,  into=c("gene_name",LETTERS[1:6]), sep="\\|" ) %>%
  select(gene_name,KO)
,

kegg %>%
  filter(!(gene_name %in% grep("Lr10",gene_name, value = TRUE))) %>%
  mutate(gene_name = gsub("\\|MetaHIT","", gene_name) ) %>%
  select(gene_name,KO)

) %>% merge(

#merge

rbind(

tolerance_MSP_table %>%
  #filter(gene_name %in% grep("Lr10",gene_name, value = TRUE)) %>%
  filter(msp_name %in% c("msp_lrhamnosus")) %>%
  tidyr::separate(gene_name,  into=c("gene_name",LETTERS[1:6]), sep="\\|" ) %>%
  select(msp_name,module_name,gene_id,gene_name,gene_length)
,

tolerance_MSP_table %>%
  filter(!(msp_name %in% c("msp_lrhamnosus"))) %>%
  mutate(gene_name = gsub("\\|MetaHIT","", gene_name))

),

by="gene_name"


) %>%

filter(msp_name %in% c("msp_1025","msp_lrhamnosus"))


```


Import CD-HIT clustering
```{r}

cdhit_path = system.file("data-raw/msp2018_tolerance/Lifeseq_non-redundant_augmented-catalog.fasta.clstr",package="Tolerance")


cdhit = vroom::vroom(cdhit_path)

Lacto_genes_name = grep("Lr10|Lpp121|Lpp1", cdhit$id, value = TRUE)


clstr_id = cdhit %>%  filter(id %in% Lacto_genes_name  ) %>%  pull(clstr) %>% unique()


corresponding_catalog_Lacto_genes = 
cdhit %>%
  filter(clstr_rep == 1) %>%
  filter(clstr %in% clstr_id) %>%
  pull(id)
  
table(gsub("\\|MetaHIT","",corresponding_catalog_Lacto_genes) %in% counts_scaled$gene_name) 



c(counts_scaled$gene_name[counts_scaled$gene_name %in%  gsub("\\|MetaHIT","",corresponding_catalog_Lacto_genes) ],

grep("Lr10|Lpp121|Lpp1",counts_scaled$gene_name, value = TRUE)) -> genes_names_from_Lacto

counts_scaled %>%
  filter(gene_name %in% genes_names_from_Lacto) %>% dim


```



```{r}

# counts_scaled %>% 
#   dplyr::rename("gene_name" = 1) %>%
#   #sample_frac(0.05) %>%
#   #filter(gene_name %in% grep("Lr10",gene_name, value = TRUE)) %>%
#   merge(tolerance_MSP_table, by="gene_name", all.x = TRUE) %>%
#   #filter(msp_name %in% c("msp_1025") | gene_name %in% grep("Lpp121|Lpp1",gene_name, value = TRUE) ) %>%
#   #filter(msp_name %in% c("msp_lrhamnosus") & gene_name %in% grep("Lpp121|Lpp1",gene_name, value = TRUE) ) %>%
#   filter(msp_name %in% c("msp_1025","msp_lrhamnosus") | gene_name %in% grep("Lpp121|Lpp1|Lr10",gene_name, value = TRUE)) %>% 
#  
#   select(msp_name,gene_name) -> genes_names_from_products

genes_names_from_products <- 
  unique(tolerance_MSP_table %>% 
  filter(msp_name %in% c("msp_1025","msp_lrhamnosus") | gene_name %in% grep("Lpp121|Lpp1|Lr10",gene_name, value = TRUE)) %>%
  .$gene_name,

counts_scaled %>% 
  dplyr::rename("gene_name" = 1) %>%
  select(gene_name) %>%
  filter(gene_name %in% grep("Lpp121|Lpp1|Lr10",gene_name, value = TRUE)) %>% .$gene_name) 



# genes_names_from_products %>%
#  tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|", fill="left") %>%
#   #select(-A,-B,-C,-D,-E,-F) %>%
#   select(-F) %>%
#   tidyr::unite("gene_names",gene_name, A,B,C,D,E, sep = "|") %>%
#   mutate(gene_name = gene_names %>% gsub("NA\\|NA\\|NA\\|NA\\|NA\\|","",.)) %>%
#   select(-gene_names) -> genes_names_from_products


```




```{r}



kegg %>% #head(200) %>%
  #sample_frac(0.01) %>%
  tidyr::separate(KO, sep=",", into=as.character(1:100), extra="drop") %>%
  # tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|", fill="left") %>%
  # #select(-A,-B,-C,-D,-E,-F) %>%
  # select(-F) %>%
  # tidyr::unite("gene_names",gene_name, A,B,C,D,E, sep = "|") %>%
  # mutate(gene_name = gene_names %>% gsub("NA\\|NA\\|NA\\|NA\\|NA\\|","",.)) %>%
  # select(-gene_names) %>% 
  reshape2::melt(id.vars="gene_name") %>%
  select(gene_name,value) %>%
  dplyr::rename(KO=value) %>%
  na.omit() %>%
  as_tibble() -> kegg_melt



kegg_melt %>% dim




```




## aggregate count at KEGG levels
```{r}

kegg_melt %>% #sample_frac(0.05) %>%

  merge(.,
        counts_scaled, #%>% #sample_frac(0.05) %>% 
          # dplyr::rename(gene_name = 1) %>% 
          # tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|", fill="left") %>%
          # #select(-A,-B,-C,-D,-E,-F) %>%
          #  select(-F) %>%
          #  tidyr::unite("gene_names",gene_name, A,B,C,D,E, sep = "|") %>%
          #  mutate(gene_name = gene_names %>% gsub("NA\\|NA\\|NA\\|NA\\|NA\\|","",.)) %>%
          #  select(-gene_names) , 
        by="gene_name") %>%
  select(-gene_name) %>%
  group_by(KO) %>%
  summarise_all(sum) -> KO_counts

write.csv2(KO_counts, file="KO_counts.csv")

KO_counts %>% head

KO_counts %>% dim



```


```{r}

kegg_melt %>% #sample_frac(0.05) %>%

  merge(.,
        counts_scaled %>% #head %>% 
          #dplyr::rename(gene_name = 1) %>% 
          # tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|", fill="left") %>%
          # #select(-A,-B,-C,-D,-E,-F) %>%
          #  select(-F) %>%
          #  tidyr::unite("gene_names",gene_name, A,B,C,D,E, sep = "|") %>%
          #  mutate(gene_name = gene_names %>% gsub("NA\\|NA\\|NA\\|NA\\|NA\\|","",.)) %>%
          #  select(-gene_names) %>%
           filter(gene_name %in% genes_names_from_Lacto), 
        by="gene_name") %>%
  select(-gene_name) %>%
  group_by(KO) %>%
  summarise_all(sum) -> KO_counts_Lactobacillus

write.csv2(KO_counts_Lactobacillus, file="KO_counts_Lactobacillus.csv")

KO_counts_Lactobacillus %>% head




```


```{r}

KO_counts %>%
  reshape2::melt() %>%
  merge(
          KO_counts_Lactobacillus %>% reshape2::melt(), 
          by=c("KO","variable")
          ) %>%
  mutate(value = value.y/(value.x+1)) %>%
  arrange(desc(value)) -> KO_contrib_per_individual


#save(KO_contrib_per_individual, file="KO_contrib_per_individual.rda")


KO_contrib_per_individual %>%
  filter(value.y>100) %>%
  arrange(desc(value))


KO_contrib_per_individual %>%
  #group_by(KO) %>%
  #summarise(value=mean(value)) %>%
  #arrange(value) %>%
  #filter(value.y>10) %>%
  #filter(KO %in% c("K03043","K04043")) %>%
  ggplot() + geom_point(aes(y=value.x,x=variable)) + scale_y_log10()
  

#to DO : cross with group & visit


#K03043
#K04043


```

```{r}

ciccarelli=c("K06942",
"K01889",
"K02950",
"K02992",
"K02967",
"K02867",
"K02863",
"K02906",
"K02890",
"K02982",
"K02874",
"K02931",
"K02994",
"K02940",
"K02988",
"K02952",
"K02948",
"K02871",
"K02996",
"K01875",
"K02953",
"K02961",
"K02866",
"K02876",
"K03076",
"K03040",
"K02881",
"K01869",
"K02987",
"K01409",
"K01873")




```




```{r fig.height=5, fig.width=9}

# sample_metadata_metaG %>%
#   filter(Subject!="276001094") %>% #exclude contaminated CP3
#   merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
#   filter(Visit %in% c("V2") | group == "V4_CP.3") %>%
#   group_by(seq) %>%
#   mutate(sum=sum(value.x)) %>%
#   ungroup() %>%
#   mutate(prop.x=value.x/sum) %>%
#   filter(prop.x> 0.0001) %>%
#   ggplot() + geom_point(aes(x=value+10^-6,y=prop.x+10^-6)) + scale_x_log10() + scale_y_log10()
# 
# 
# sample_metadata_metaG %>%
#   filter(Subject!="276001094") %>% #exclude contaminated CP3
#   merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
#   filter(group == "V4_ATN.3") %>%
#   group_by(seq) %>%
#   mutate(sum=sum(value.x)) %>%
#   ungroup() %>%
#   mutate(prop.x=value.x/sum) %>%
#   filter(prop.x> 0.0001) %>%
#   ggplot() + geom_point(aes(x=value+10^-6,y=prop.x+10^-6)) + scale_x_log10() + scale_y_log10()

  
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
  #filter(group == "V4_ATN.3") %>%
  #filter(Visit != "V4") %>%
  group_by(seq) %>%
  mutate(sum_kegg=sum(value.x)) %>%
  ungroup() %>%
  mutate(prop.x=value.x/sum_kegg, prop.y=value.y/sum_kegg) %>%
  #filter(prop.x> 0.0001) %>%
  ggplot() + geom_point(aes(x=prop.y+10^-7,y=prop.x+10^-7, col=group), alpha=0.5) + 
  scale_x_log10() + scale_y_log10() +
  xlab("FMP genes rel. abundance contributions\nto KEGG Orthologous Group") +
  ylab("KEGG Orthologous Group relative abundance") + facet_grid(Visit~Groupe) +
  geom_vline(xintercept = 0.000001)


KO_DNA_replication2 = read.table("../data-raw/KEGG_DNA_replication/KEGG_KO_DNA_replication.txt", sep="\t") %>%
  pull(V1) %>% gsub("\xa0\xa0","",.)

KO_DNA_replication3 = kegg_modules %>% filter(ModuleGroup2 %in% c("DNA polymerase","RNA polymerase")) %>% pull(KO)


KO_DNA_replication = ciccarelli



                    
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
  #filter(group == "V4_ATN.3") %>%
  #filter(Visit != "V4") %>%
  group_by(seq) %>%
  mutate(sum_kegg=sum(value.x)) %>%
  ungroup() %>%
  mutate(prop.x=value.x/sum_kegg, prop.y=value.y/sum_kegg) %>%
  mutate(DNA_KO = ifelse(KO %in% KO_DNA_replication, "universal orthologous genes","others")) %>%
  ggplot() + geom_point(aes(x=prop.y+10^-7,y=prop.x+10^-7), alpha=0.5) + 
  scale_x_log10() + scale_y_log10() +
  xlab("FMP genes rel. abundance contributions\nto KEGG Orthologous Group") +
  ylab("KEGG Orthologous Group relative abundance") + facet_grid(Visit~Groupe+DNA_KO) +
  geom_vline(xintercept = 0.000003) +
  geom_hline(yintercept = 0.0001)


sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
  #filter(group == "V4_ATN.3") %>%
  #filter(Visit != "V4") %>%
  group_by(seq) %>%
  mutate(sum_kegg=sum(value.x)) %>%
  ungroup() %>%
  mutate(prop.x=value.x/sum_kegg, prop.y=value.y/sum_kegg) %>%
  mutate(DNA_KO = ifelse(KO %in% KO_DNA_replication, "Ciccarelli universal genes","others genes")) %>%
  ggplot() + geom_point(aes(x=prop.y+10^-7,y=prop.x+10^-7, color=DNA_KO, alpha=DNA_KO, size=DNA_KO)) + 
  scale_x_log10() + scale_y_log10() +
  scale_size_discrete("KEGG pathway",range=c(3,1)) +
  scale_alpha_discrete("KEGG pathway",range=c(1,0.05)) +
  scale_color_manual("KEGG pathway",values = c("red","lightblue")) +
  xlab("FMP genes rel. abundance contributions\nto KEGG Orthologous group") +
  ylab("KEGG Orthologous group\naggregated relative abundance") + facet_wrap(~Groupe+Visit, nrow = 2) +
  geom_vline(xintercept = 0.000003, linetype=2) +
  geom_hline(yintercept = 0.0001, linetype=2) + theme_minimal()

ggsave("KEGG_contribution_per_visit_per_product.pdf")


# sample_metadata_metaG %>%
#   filter(Subject!="276001094") %>% #exclude contaminated CP3
#   merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
#   #filter(group == "V4_ATN.3") %>%
#   #filter(Visit != "V4") %>%
#   group_by(seq) %>%
#   mutate(sum_kegg=sum(value.x)) %>%
#   ungroup() %>%
#   mutate(prop.x=value.x/sum_kegg, prop.y=value.y/sum_kegg) %>%
#   #filter(prop.x> 0.0001) %>%
#   ggplot() + geom_point(aes(x=value+10^-6,y=prop.x+10^-6, col=group), alpha=0.5) + 
#   scale_x_log10() + scale_y_log10() +
#   xlab("FMP genes rel. abundance contributions\nto KEGG Orthologous Group") +
#   ylab("KEGG Orthologous Group relative abundance") + facet_grid(Visit~Groupe)


# sample_metadata_metaG %>%
#   filter(Subject!="276001094") %>% #exclude contaminated CP3
#   merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
#   #filter(group == "V4_ATN.3") %>%
#   #filter(Visit != "V4") %>%
#   group_by(seq) %>%
#   mutate(sum_kegg=sum(value.x)) %>%
#   ungroup() %>%
#   mutate(prop.x=value.x/sum_kegg, prop.y=value.y/sum_kegg) %>%
#   filter(value> 0.0001)


```




```{r}
kegg_contrib_V2=
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
  #filter(group == "V4_ATN.3") %>%
  filter(Visit == "V2") %>%
  group_by(seq) %>%
  mutate(sum_kegg=sum(value.x)) %>%
  ungroup() %>%
  mutate(prop.x=value.x/sum_kegg, prop.y=value.y/sum_kegg)

kegg_contrib_V4=
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
  #filter(group == "V4_ATN.3") %>%
  filter(Visit == "V4") %>%
  group_by(seq) %>%
  mutate(sum_kegg=sum(value.x)) %>%
  ungroup() %>%
  mutate(prop.x=value.x/sum_kegg, prop.y=value.y/sum_kegg)



KO_DNA_replication = read.table("../data-raw/KEGG_DNA_replication/KEGG_KO_DNA_replication.txt", sep="\t") %>%
  pull(V1) %>% gsub("\xa0\xa0","",.)
                    
KO_DNA_replication = ciccarelli

kegg_contrib_V2 %>% #sample_frac(0.1) %>%
  merge(kegg_contrib_V4, by=c("Subject","KO")) %>%
  mutate(delta_contrib = value.y-value.x) %>%
  filter(KO %in% KO_counts_Lactobacillus$KO) %>%
  group_by(Groupe.x,KO) %>%
  summarise(mean_contrib=median(delta_contrib+10^-6), mean_KO=median(prop.x.x+10^-6)) %>%
  mutate(DNA_KO = ifelse(KO %in% KO_DNA_replication, "universal genes","others genes")) %>%
  ggplot() + geom_point(aes(x=mean_KO,y=mean_contrib, color=Groupe.x), alpha=0.5) + 
  scale_y_log10(label=scales::percent) + scale_x_log10() + facet_wrap(~DNA_KO) +
  geom_hline(yintercept = 0.005, linetype=2) +
  scale_color_brewer("", type="qual") +
  xlab("average KO aggregated rel. abundance\nat baseline") + 
  ylab("average Test product gene\ncontribution by KO\nduring consumption") +
  theme_minimal()
  
ggsave("Figure_KEGG_average_contribution.pdf")




```


```{r fig.height=5, fig.width=10}
kegg_modules = readr::read_tsv("../data-raw/kegg.module.txt")

kegg_modules_plot=
kegg_contrib_V2 %>% #sample_frac(0.1) %>%
  merge(kegg_contrib_V4, by=c("Subject","KO")) %>%
  mutate(delta_contrib = value.y-value.x) %>%
  filter(KO %in% KO_counts_Lactobacillus$KO) %>%
  group_by(Groupe.x,KO) %>%
  summarise(mean_contrib=median(delta_contrib+10^-6), mean_KO=median(prop.x.x+10^-6)) %>%
  ungroup() %>%
  filter(mean_contrib>0.005) %>%
  merge(kegg_modules, by="KO", all.x = TRUE) %>%
  group_by(KO) %>%
  slice(1) %>%
  ungroup() %>%
  arrange(desc(mean_contrib)) %>%
  dplyr::top_n(50, mean_contrib) %>%
  #mutate(ModuleGroup2 = ifelse(is.na(ModuleGroup2),"Unknown modules",ModuleGroup2) ) %>%
  ggplot() + geom_bar(aes(x=reorder(KO, -mean_contrib) ,y=mean_contrib*100, fill=ModuleGroup2), stat="identity") +
  ylab("average Test product gene\ncontribution (%) by KO\nafter consumption") +
  xlab("KEGG Orthologous group") +
  theme_minimal() +
  theme(axis.text.x=element_text(angle=90,hjust=1)) +
  scale_y_log10() +
  scale_fill_discrete("KEGG modules", l = 80, c = 150)
  
kegg_modules_plot  
ggsave("Figure_kegg_barplot.pdf")


  
```

```{r}


kegg_contrib_V2 %>% #sample_frac(0.1) %>%
  merge(kegg_contrib_V4, by=c("Subject","KO")) %>%
  mutate(delta_contrib = value.y-value.x) %>%
  filter(KO %in% KO_counts_Lactobacillus$KO) %>%
  group_by(Groupe.x,KO) %>%
  summarise(mean_contrib=median(delta_contrib+10^-6), mean_KO=median(prop.x.x+10^-6)) %>%
  ungroup() %>%
  filter(mean_contrib>0.005) %>%
  select(KO,mean_contrib) %>%
  write.csv2(file = "KO_selected_FMP_contribution.csv")


```


```{r}
rbind(  cbind(readxl::read_xlsx("../data-raw/KEGG_annotations/KO_selected_FMP_contribution.xlsx", sheet = "transporter"),
    Brite="Transporters"),
  cbind(    readxl::read_xlsx("../data-raw/KEGG_annotations/KO_selected_FMP_contribution.xlsx", sheet = "enzymes"),
    Brite="Enzymes")) %>%
  select(KO,Brite,Type,Subtype) %>% 
  mutate(Type = case_when(
    
    Type == "AA" ~ "Aminoacid transporter",
    Type == "PTS" ~ "Phosphotransferase system",
    Type == "ion" ~ "Ion transporter",
    Type == "MFS" ~ "MFS transporter",
    Type == "EC:1" ~ "Oxydoreductase",
    Type == "EC:2" ~ "Transferase",
    Type == "EC.2" ~ "Transferase", # to fix
    Type == "EC:3" ~ "Hydrolase",
    Type == "EC:4" ~ "Lyase",
    Type == "EC:5" ~ "Isomerase",
    Type == "EC:6" ~ "Ligase",
    TRUE ~ "Others"       
           
           )) %>%
  mutate(Subtype = case_when(
    
    Subtype == "EC:1.1" ~ "EC:1.1",
    Subtype == "EC:2.7" ~ "EC:2.7",
    Subtype == "EC:3.5" ~ "EC:3.5",
    Subtype == "DHA" ~ "Drug antiporter",
    Subtype == "manganese" ~ "Manganese transporter",
    Subtype == "SP" ~ "Sugar porter",
    Subtype == "glu" ~ "Glutamate transporter",
    TRUE ~ "Others"
    
    )) %>%
  mutate(Type=factor(Type, levels=c("Others","MFS transporter", "Aminoacid transporter","Phosphotransferase system","Ion transporter", "Transferase", "Oxydoreductase","Hydrolase","Lyase","Isomerase","Ligase"))) %>%
  mutate(Subtype=factor(Subtype, levels=c("Others", "Glutamate transporter","Manganese transporter","Drug antiporter","Sugar porter","EC:1.1","EC:2.7","EC:3.5"))) %>%
  ggplot() +
  geom_bar(aes(x=Type,fill=Subtype)) + 
  facet_wrap(~Brite, scale="free_y", nrow=2) +
  ylab("Number of KOs")+ xlab("Category") +
  theme_minimal() +
  scale_fill_manual("", values= c('#cccccc','#e41a1c','#377eb8','#4daf4a','#984ea3','#ff7f00','#ffff33','#a65628')) +
  coord_flip() -> kegg_brite_barplot
  
 kegg_brite_barplot 
  


```



```{r eval=FALSE, include=FALSE}

#import function category based on KEGG brite

#readxl::read_xlsx("../data-raw/KEGG_annotations/KO_selection_FMP_category.xlsx", sheet = "category") %>% readr::write_tsv("KO_category.txt")
#readr::read_tsv("KO_category.txt") %>%
#mutate(KO = KO %>% stringr::str_remove(., "[:space:][:space:]") %>% stringr::str_remove(., "ko:"))

readxl::read_xlsx("../data-raw/KEGG_annotations/KO_selection_FMP_category.xlsx", sheet = "category") %>%
  mutate(KO = KO %>% stringr::str_remove(., "[:space:][:space:]") %>% stringr::str_remove(., "ko:")) %>%
  mutate(KEGG_Brite = ifelse(KEGG_Brite=="others","Others",KEGG_Brite)) %>%
  group_by(KEGG_Brite,Type) %>%
  summarise(n=n()) %>%
  arrange(desc(n)) %>%
  mutate(Type= ifelse(Type %in% c("AA","EC: 1.1","PTS","EC: 2.7","EC: 3.5","ion"),yes=Type,no=NA)) %>%
  #mutate(Type = forcats::fct_relevel(Type,"AA","PTS","ion","EC: 1.1","EC: 2.7","EC: 3.5")) %>%
  mutate(Type=factor(Type, levels=c("AA","PTS","ion","EC: 1.1","EC: 2.7","EC: 3.5"))) %>%
  #mutate(Type = forcats::fct_relevel(as.factor(Type),"AA",after=3)) %>%
  mutate(Type = case_when(
    
    Type == "AA" ~ "Aminoacid transporter",
    Type == "PTS" ~ "Phosphotransferase system",
    Type == "ion" ~ "Ion transporter",
    Type == "EC: 1.1" ~ "EC:1.1",
    Type == "EC: 2.7" ~ "EC:2.7",
    Type == "EC: 3.5" ~ "EC:3.5"
    
  )) %>%
  mutate(Type=factor(Type, levels=c("Aminoacid transporter","Phosphotransferase system","Ion transporter","EC:1.1","EC:2.7","EC:3.5"))) %>%
  mutate(Type = forcats::fct_explicit_na(Type, na_level = "Others") ) %>%
  mutate(Type = Type %>% forcats::fct_relevel("Others", after = 0)) %>%
  ungroup() %>%
  #mutate(KEGG_Brite = forcats::fct_reorder(KEGG_Brite %>% as.factor, n, median)) %>%
  ggplot() + 
  geom_bar(aes(x=reorder(KEGG_Brite,-n),fill=Type,y=n), stat="identity") + 
  xlab("") + ylab("Number of KO") +
  coord_flip() + theme_classic() +
  scale_fill_manual("", values= c('#cccccc','#e41a1c','#377eb8','#4daf4a','#984ea3','#ff7f00','#ffff33')) -> kegg_brite_barplot
  
 kegg_brite_barplot 



```




```{r eval=FALSE, include=FALSE}

kegg_contrib_V2 %>% #sample_frac(0.1) %>%
  merge(kegg_contrib_V4, by=c("Subject","KO")) %>%
  mutate(delta_contrib = value.y-value.x) %>%
  filter(KO %in% KO_counts_Lactobacillus$KO) %>%
  group_by(Groupe.x,KO) %>%
  summarise(mean_contrib=median(delta_contrib+10^-6), mean_KO=median(prop.x.x+10^-6)) %>%
  ungroup() %>%
  filter(mean_contrib>0.005) %>%
  merge(kegg_modules, by="KO", all.x = TRUE) %>%
  select(KO,KODesc) %>%
  unique() %>% readr::write_csv2(path="KO_selection_FMP.csv")




```

```{r}

kegg_contrib_V2 %>% #sample_frac(0.1) %>%
  merge(kegg_contrib_V4, by=c("Subject","KO")) %>%
  mutate(delta_contrib = value.y-value.x) %>%
  filter(KO %in% KO_counts_Lactobacillus$KO) %>%
  group_by(Groupe.x,KO) %>%
  summarise(mean_contrib=median(delta_contrib+10^-6), mean_KO=median(prop.x.x+10^-6)) %>%
  mutate(DNA_KO = ifelse(KO %in% KO_DNA_replication, "DNA RNA machinery","others")) %>%
  filter(mean_contrib > 0.005) %>%
  ungroup %>%
  select(KO,mean_contrib) -> KO_contrib_Lactobacillus
  write.csv2(KO_contrib_Lactobacillus, file="KO_contrib_Lactobacillus.csv")


```

```{r}
KO_contrib_Lactobacillus %>% dim


  
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
  filter(KO %in% KO_contrib_Lactobacillus$KO) %>%
  #filter(group == "V4_ATN.3") %>%
  #filter(value>0.01) %>%
  ggplot() + geom_boxplot(aes(x=KO, y=value)) + facet_wrap(~group) + scale_y_log10()
  

#to do geom_tile


sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
  mutate(contribution=ifelse(KO %in% KO_contrib_Lactobacillus$KO,"high","low")) %>%
  ggplot() + geom_tile(aes(x=KO, y=Subject, fill=value+10^-6)) + 
  facet_grid(group~contribution, scales = "free", space="free") + 
  theme(axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    axis.text.y=element_blank(),
    axis.ticks.y=element_blank()) +
  scale_fill_viridis_c(name="FMP contribution\nto KO rel. abundance",trans = "log10", label=scales::percent, breaks = c(0.0001,0.001,0.01,0.1))



```

### generate kegg ipath input
```{r}

sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(KO_contrib_per_individual, by.x="seq", by.y="variable") %>%
  filter(Groupe=="ATN" & Visit =="V4") %>%
  group_by(KO) %>%
  summarise(FMP_contrib=median(value)) %>% #pull(contribution_median) -> test
  mutate(contribution=ifelse(KO %in% KO_contrib_Lactobacillus$KO,"w20","w10")) %>%
  mutate(color=viridis::viridis(n=256)[round((((log10(FMP_contrib) - log10(min(FMP_contrib))) / (max(log10(FMP_contrib)) - min(log10(FMP_contrib)))) * 255)+1)]) %>%
  mutate(color=paste0("c",color)) %>%
  select(KO,color,contribution) %>%
  readr::write_tsv("FMP_KEGG_contribution.tsv", col_names = FALSE)





```



## Cazy


```{r eval=FALSE, include=FALSE}
cazy %>% #head(200) %>%
  #sample_frac(0.05) %>%
  tidyr::separate(cazy_family , sep=",", into=as.character(1:100)) %>%
  #tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|") %>%
  #select(-A,-B,-C,-D,-E,-F) %>%
  reshape2::melt(id.vars="gene_name") %>%
  select(gene_name,value) %>%
  dplyr::rename(cazy_family =value) %>%
  na.omit() %>%
  as_tibble() -> cazy_melt




cazy_melt %>% #sample_frac(0.05) %>%

  merge(.,
        counts_scaled, #%>% #head %>% 
          # dplyr::rename(gene_name = 1) %>% 
          # tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|") %>%
          # select(-A,-B,-C,-D,-E,-F) , 
        by="gene_name") %>%
  select(-gene_name) %>%
  group_by(cazy_family ) %>%
  summarise_all(sum) -> cazy_counts

write.csv2(cazy_counts, file="cazy_counts.csv")

cazy_counts %>% head



cazy_melt %>% #sample_frac(0.05) %>%

  merge(.,
        counts_scaled %>% #head %>% 
          dplyr::rename(gene_name = 1) %>% 
          filter(gene_name %in% genes_names_from_products), #%>%
          # tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|") %>%
          # select(-A,-B,-C,-D,-E,-F) , 
        by="gene_name") %>%
  select(-gene_name) %>%
  group_by(cazy_family) %>%
  summarise_all(sum) -> cazy_counts_Lactobacillus

write.csv2(cazy_counts_Lactobacillus, file="cazy_counts_Lactobacillus.csv")





```



```{r eval=FALSE, include=FALSE}

cazy_counts %>%
  reshape2::melt() %>%
  merge(
          cazy_counts_Lactobacillus %>% reshape2::melt(), 
          by=c("cazy_family","variable")
          ) %>%
  mutate(value = value.y/(value.x+1)) %>%
  arrange(desc(value)) -> cazy_contrib_per_individual


cazy_contrib_V2=
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(cazy_contrib_per_individual, by.x="seq", by.y="variable") %>%
  #filter(group == "V4_ATN.3") %>%
  filter(Visit == "V2") %>%
  group_by(seq) %>%
  mutate(sum_cazy=sum(value.x)) %>%
  ungroup() %>%
  mutate(prop.x=value.x/sum_cazy, prop.y=value.y/sum_cazy)

cazy_contrib_V4=
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(cazy_contrib_per_individual, by.x="seq", by.y="variable") %>%
  #filter(group == "V4_ATN.3") %>%
  filter(Visit == "V4") %>%
  group_by(seq) %>%
  mutate(sum_cazy=sum(value.x)) %>%
  ungroup() %>%
  mutate(prop.x=value.x/sum_cazy, prop.y=value.y/sum_cazy)



cazy_contrib_V2 %>% #sample_frac(0.1) %>%
  merge(cazy_contrib_V4, by=c("Subject","cazy_family")) %>%
  mutate(delta_contrib = value.y-value.x) %>%
  filter(cazy_family %in% cazy_counts_Lactobacillus$cazy_family) %>%
  group_by(Groupe.x,cazy_family) %>%
  summarise(mean_contrib=median(delta_contrib+10^-6), mean_cazy_family=median(prop.x.x+10^-6)) %>%
  #mutate(DNA_KO = ifelse(KO %in% KO_DNA_replication, "DNA RNA machinery","others")) %>%
  filter(mean_contrib > 0.003) %>%
  select(cazy_family,mean_contrib) -> cazy_contrib_Lactobacillus
  write.csv2(cazy_contrib_Lactobacillus, file="cazy_contrib_Lactobacillus.csv")





```


```{r eval=FALSE, include=FALSE}
cazy_contrib_Lactobacillus %>% dim


  
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(cazy_contrib_per_individual, by.x="seq", by.y="variable") %>%
  filter(cazy_family %in% cazy_contrib_Lactobacillus$cazy_family) %>%
  #filter(group == "V4_ATN.3") %>%
  #filter(value>0.01) %>%
  ggplot() + geom_boxplot(aes(x=cazy_family, y=value)) + facet_wrap(~group) + scale_y_log10()
  

#to do geom_tile


sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(cazy_contrib_per_individual, by.x="seq", by.y="variable") %>%
  mutate(contribution=ifelse(cazy_family %in% cazy_contrib_Lactobacillus$cazy_family,"high","low")) %>%
  ggplot() + geom_tile(aes(x=cazy_family, y=Subject, fill=value+10^-6)) + 
  facet_grid(group~contribution, scales = "free", space="free") + 
  theme(axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    axis.text.y=element_blank(),
    axis.ticks.y=element_blank()) +
  scale_fill_viridis_c(name="FMP contribution\nto cazy rel. abundance",trans = "log10", label=scales::percent, breaks = c(0.0001,0.001,0.01,0.1))



```




## EggNOG


```{r eval=FALSE, include=FALSE}

og %>% head(50)

#NOG = og$OG %>% strsplit(split=",") %>% sapply(., function(x){grep("@NOG",x , value=TRUE) %>% paste(collapse = ",")})



```



```{r eval=FALSE, include=FALSE}

OG=og

OG %>% #head(200) %>%
  #sample_frac(0.05) %>%
  #tidyr::separate(OG , sep=",", into=as.character(1:100)) %>%
  tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|") %>%
  select(-A,-B,-C,-D,-E,-F) %>%
  reshape2::melt(id.vars="gene_name") %>%
  select(gene_name,value) %>%
  dplyr::rename(OG =value) %>%
  na.omit() %>%
  as_tibble() -> OG_melt




OG_melt %>% #sample_frac(0.05) %>%

  merge(.,
        counts_scaled %>% #head %>% 
          dplyr::rename(gene_name = 1) %>% 
          tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|") %>%
          select(-A,-B,-C,-D,-E,-F) , 
        by="gene_name") %>%
  select(-gene_name) %>%
  group_by(OG ) %>%
  summarise_all(sum) -> OG_counts


NOG = OG_counts$OG %>% strsplit(split=",") %>% sapply(., function(x){grep("@NOG",x , value=TRUE) %>% paste(collapse = ",")})

OG_counts$NOG = NOG

OG_counts %>%
  select(-OG) %>%
  group_by(NOG) %>%
  summarise_all(sum) -> NOG_counts


write.csv2(OG_counts, file="OG_counts.csv")
write.csv2(NOG_counts, file="NOG_counts.csv")


NOG_counts %>% head



OG_melt %>% #sample_frac(0.05) %>%

  merge(.,
        counts_scaled %>% #head %>% 
          dplyr::rename(gene_name = 1) %>% 
          filter(gene_name %in% genes_names_from_products$gene_name) %>%
          tidyr::separate(gene_name, into=c("gene_name",LETTERS[1:6]), sep="\\|") %>%
          select(-A,-B,-C,-D,-E,-F) , 
        by="gene_name") %>%
  select(-gene_name) %>%
  group_by(OG) %>%
  summarise_all(sum) -> OG_counts_Lactobacillus

NOG_lact = OG_counts_Lactobacillus$OG %>% strsplit(split=",") %>% sapply(., function(x){grep("@NOG",x , value=TRUE) %>% paste(collapse = ",")})

OG_counts_Lactobacillus$NOG = NOG_lact

OG_counts_Lactobacillus %>%
  select(-OG) %>%
  group_by(NOG) %>%
  summarise_all(sum) -> NOG_counts_Lactobacillus

write.csv2(NOG_counts_Lactobacillus, file="NOG_counts_Lactobacillus.csv")





```



```{r eval=FALSE, include=FALSE}

NOG_counts %>%
  reshape2::melt() %>%
  merge(
          NOG_counts_Lactobacillus %>% reshape2::melt(), 
          by=c("NOG","variable")
          ) %>%
  mutate(value = value.y/(value.x+1)) %>%
  arrange(desc(value)) -> NOG_contrib_per_individual


NOG_contrib_V2=
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(NOG_contrib_per_individual, by.x="seq", by.y="variable") %>%
  #filter(group == "V4_ATN.3") %>%
  filter(Visit == "V2") %>%
  group_by(seq) %>%
  mutate(sum_NOG=sum(value.x)) %>%
  ungroup() %>%
  mutate(prop.x=value.x/sum_NOG, prop.y=value.y/sum_NOG)

NOG_contrib_V4=
sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(NOG_contrib_per_individual, by.x="seq", by.y="variable") %>%
  #filter(group == "V4_ATN.3") %>%
  filter(Visit == "V4") %>%
  group_by(seq) %>%
  mutate(sum_NOG=sum(value.x)) %>%
  ungroup() %>%
  mutate(prop.x=value.x/sum_NOG, prop.y=value.y/sum_NOG)



NOG_contrib_V2 %>% #sample_frac(0.1) %>%
  merge(NOG_contrib_V4, by=c("Subject","NOG")) %>%
  mutate(delta_contrib = value.y-value.x) %>%
  filter(NOG %in% NOG_counts_Lactobacillus$NOG) %>%
  group_by(Groupe.x,NOG) %>%
  summarise(mean_contrib=median(delta_contrib+10^-6), mean_NOG_family=median(prop.x.x+10^-6)) %>% 
  #mutate(DNA_KO = ifelse(KO %in% KO_DNA_replication, "DNA RNA machinery","others")) %>%
  filter(mean_contrib > 0.01) %>%
  select(NOG,mean_contrib) -> NOG_contrib_Lactobacillus
  write.csv2(NOG_contrib_Lactobacillus, file="NOG_contrib_Lactobacillus.csv")





```


```{r eval=FALSE, include=FALSE}
NOG_contrib_Lactobacillus %>% dim


#   
# sample_metadata_metaG %>%
#   filter(Subject!="276001094") %>% #exclude contaminated CP3
#   merge(cazy_contrib_per_individual, by.x="seq", by.y="variable") %>%
#   filter(cazy_family %in% cazy_contrib_Lactobacillus$cazy_family) %>%
#   #filter(group == "V4_ATN.3") %>%
#   #filter(value>0.01) %>%
#   ggplot() + geom_boxplot(aes(x=cazy_family, y=value)) + facet_wrap(~group) + scale_y_log10()
#   

#to do geom_tile


sample_metadata_metaG %>%
  filter(Subject!="276001094") %>% #exclude contaminated CP3
  merge(NOG_contrib_per_individual, by.x="seq", by.y="variable") %>%
  mutate(contribution=ifelse(NOG %in% NOG_contrib_Lactobacillus$NOG,"high","low")) %>%
  ggplot() + geom_tile(aes(x=NOG, y=Subject, fill=value+10^-6)) + 
  facet_grid(group~contribution, scales = "free", space="free") + 
  theme(axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    axis.text.y=element_blank(),
    axis.ticks.y=element_blank()) +
  scale_fill_viridis_c(name="FMP contribution\nto OGs rel. abundance",trans = "log10", label=scales::percent, breaks = c(0.0001,0.001,0.01,0.1))



```



## MSP functional redundancy


```{r}
data("MSP_abundance_matrix_rounded")

MSP_abundance_matrix_rounded %>% 
  tibble::rownames_to_column("msp") %>% 
  mutate_if(is.numeric, function(x) x/sum(x)) %>% 
  tibble::column_to_rownames("msp") %>%
  BiotypeR::noise.removal(percent=1) %>% rownames() -> msp_select

msp_select = union(msp_select,c("msp_lrhamnosus","msp_1025"))

#read_mass
sum(MSP_abundance_matrix_rounded[msp_select,])/sum(MSP_abundance_matrix_rounded)


```



### with CAZY


```{r eval=FALSE, include=FALSE}

cazy_melt %>% 
  filter(cazy_family %in% cazy_contrib_Lactobacillus$cazy_family) %>% 
  merge(tolerance_MSP_table, by="gene_name") %>%
  select(cazy_family, msp_name) %>% 
  unique() %>% 
  group_by(msp_name) %>%
  summarise(n=n()) %>%
  merge(MSP_tax, by.x="msp_name", by.y="MSP_ID") %>%
  arrange(desc(n)) %>%
  filter(msp_name %in% msp_select)

  
  

```


### with KEGG


```{r}

kegg_melt %>% 
  filter(KO %in% KO_contrib_Lactobacillus$KO) %>% 
  merge(tolerance_MSP_table, by="gene_name") %>% #filter(msp_name=="msp_lrhamnosus")
  select(KO, msp_name) %>% 
  unique() %>% 
  group_by(msp_name) %>%
  summarise(n=n()) -> kegg_summary


kegg_summary %>% filter(msp_name=="msp_lrhamnosus")

kegg_summary %>%
  merge(MSP_tax %>% select(MSP_ID,species) %>% rbind(c("msp_lrhamnosus","Lactobacillus rhamnosus")), by.x="msp_name", by.y="MSP_ID") %>%
  arrange(desc(n)) %>%
  filter(msp_name %in% msp_select) -> kegg_similarity

kegg_similarity %>%
  filter(msp_name=="msp_lrhamnosus")


```



## ecological MSP network


```{r}

MSP_abundance_matrix_rounded %>% 
  tibble::rownames_to_column("msp") %>% 
  mutate_if(is.numeric, function(x) x/sum(x)) %>% 
  tibble::column_to_rownames("msp") %>%
  BiotypeR::noise.removal(percent=1) %>% rownames() -> msp_select

msp_select = union(msp_select,c("msp_lrhamnosus","msp_1025"))


```


```{r eval=FALSE, include=FALSE}
#start this job as a external job

library(SpiecEasi)

MSP_abundance_matrix_rounded[msp_select,] %>% t %>%
  spiec.easi(., method='mb', lambda.min.ratio=1e-2,
                          nlambda=20, pulsar.params=list(rep.num=20, ncores=1)) -> MSP_spiec_easi


MSP_spiec_easi = run_spiec_easi_results$MSP_spiec_easi

save(MSP_spiec_easi, file = "MSP_spiec_easi.rda")


```




```{r}

library(ggraph)
library(tidygraph)
library(igraph)
load("MSP_spiec_easi.rda")




ig.mb     <- adj2igraph(getRefit(MSP_spiec_easi), vertex.attr=list(name=msp_select))

## set size of vertex proportional to clr-mean
vsize    <- rowMeans(clr(MSP_abundance_matrix_rounded[msp_select,], 1))+6
ig.coord <- layout.fruchterman.reingold(ig.mb)
ig.coord <- layout.circle(ig.mb)


plot(ig.mb, layout=ig.coord, vertex.size=vsize,  main="MB", vertex.shape="none")


igraph::plot.igraph(ig.mb)

as.matrix(ig.mb)

sebeta <- symBeta(getOptBeta(MSP_spiec_easi), mode='maxabs')

get.adjacency(ig.mb, attr="weight", sparse=F) %>%
  reshape2::melt() %>%
  mutate(w= sebeta %>% Matrix::as.matrix() %>% reshape2::melt() %>% .$value ) %>%
  #unique %>%
  reshape2:: dcast(Var1~Var2, value.var = "w") %>%
  tibble::column_to_rownames("Var1") %>%
  as.dist() %>%
  as.matrix %>%
  reshape2::melt() %>%
  filter(abs(value)> 0.01) %>%
  #filter(Var1 %in% c("msp_lrhamnosus","msp_1025", "msp_0790","msp_0027","msp_0854") | Var2 %in% c("msp_lrhamnosus","msp_1025", "msp_0790","msp_0027","msp_0854")) %>%
  merge(MSP_tax %>% select(MSP_ID,species) %>% rbind(c("msp_lrhamnosus","Lactobacillus rhamnosus")), by.x="Var1", by.y="MSP_ID") %>%
  merge(MSP_tax %>% select(MSP_ID,species) %>% rbind(c("msp_lrhamnosus","Lactobacillus rhamnosus")), by.x="Var2", by.y="MSP_ID") -> tolerance_spiec_easi_edge_network





```



```{r}
tolerance_spiec_easi_edge_network

graph <- as_tbl_graph(
  data.frame(
    from = tolerance_spiec_easi_edge_network$Var1,
    to = tolerance_spiec_easi_edge_network$Var2,
    weight = abs(tolerance_spiec_easi_edge_network$value),
    type= tolerance_spiec_easi_edge_network$value > 0
  )
)

data.frame(name=V(graph)$name) %>% 
  merge(.,kegg_similarity, by.x="name", by="msp_name", all.x = "TRUE") %>%
  tibble::column_to_rownames("name") %>%
  .[V(graph)$name,]

V(graph)$kegg = 
  data.frame(name=V(graph)$name) %>% 
  merge(.,kegg_similarity, by.x="name", by="msp_name", all.x = "TRUE") %>%
  tibble::column_to_rownames("name") %>%
  .[V(graph)$name,] %>% .$n

V(graph)$species = 
  data.frame(name=V(graph)$name) %>% 
  merge(.,kegg_similarity, by.x="name", by="msp_name", all.x = "TRUE") %>%
  tibble::column_to_rownames("name") %>%
  .[V(graph)$name,] %>% .$species




ggraph(graph) +
  geom_edge_link(aes(alpha=weight, color=type)) + 
  geom_node_point(aes(color=kegg)) +
  geom_node_text(aes(label=species), size=2) +
  scale_color_viridis(trans="log10")
                  


V(graph)$species[287]
max(V(graph)$kegg %>% na.omit)


graph %>%  
  mutate(dist_to_lr10 = node_distance_to(287)) %>% 
  activate(edges) %>%
  filter(type==TRUE) %>%
  activate(nodes) %>%
  mutate(isolated= node_is_isolated()) %>%
  filter(!(name %in% c("msp_lrhamnosus","msp_1025"))) %>%
  filter(!isolated) %>%
  mutate(kegg=ifelse(is.na(kegg),0,kegg)) %>% 
  select(kegg,dist_to_lr10) %>% as_tibble() %>% 
  with(.,cor.test(.$kegg, .$dist_to_lr10, method="spearman"))




network_plot = graph %>%  
  mutate(dist_to_lr10 = node_distance_to(287)) %>% 
  #mutate(FMP= ifelse(name %in% c("msp_lrhamnosus","msp_1025","msp_0790"),"FMP","others")) %>%
  mutate(FMP= ifelse(name %in% c("msp_lrhamnosus","msp_1025"),"FMP","others")) %>%
  #filter(dist_to_lr10 < 6) %>%
  ggraph() +
  geom_edge_link(aes(alpha=weight, color=type)) + 
  scale_edge_color_manual("sens", labels =c("neg.","pos."), values = c("red","lightblue")) +
  geom_node_point(aes(color=kegg/max(V(graph)$kegg %>% na.omit),size=11-dist_to_lr10)) +
  #geom_node_text(aes(label=species), size=2) +
  scale_color_viridis("Common KOs prop.\nwith Test product\nspecies",trans="log10") + 
  scale_size_continuous("Edge distance\nto Test product\nspecies", breaks = c(1,4,7,10), labels = c("9","6","3","0")) +
  scale_edge_alpha_continuous("SPIEC-EASI\nweight") + 
  ggforce::geom_mark_rect(aes(x, y, label = 'Test product\nspecies', 
                        #description = "L. rhamnosus\nL. casei\nS. thermophilus",
                        description = "L. rhamnosus\nL. casei",
                        filter = FMP == "FMP"))




graph %>%  
  mutate(dist_to_lr10 = node_distance_to(287)) %>% 
  activate(edges) %>%
  #filter(type==FALSE) %>%
  activate(nodes) %>%
  mutate(isolated= node_is_isolated()) %>%
  filter(!isolated) %>%
  mutate(kegg=ifelse(is.na(kegg),0,kegg)) %>%
  ggplot() + 
  geom_jitter(aes(x=kegg+1,y=dist_to_lr10, col=kegg), width = 0.01, height = 0.01, alpha=0.5) + 
  scale_color_viridis() +
  scale_x_log10() #+ scale_y_log10()



kegg_function_plot = graph %>%  
  mutate(dist_to_lr10 = node_distance_to(287)) %>% 
  activate(edges) %>%
  filter(type==TRUE) %>%
  activate(nodes) %>%
  mutate(isolated= node_is_isolated()) %>%
  filter(!isolated) %>%
  mutate(kegg=ifelse(is.na(kegg),0,kegg)) %>%
  mutate(kegg_break=case_when(
    
    kegg < 10 ~ "[0-10[",
    kegg>=10 & kegg < 30 ~ "[10-30[",
    kegg>=30 & kegg < 100 ~ "[30-100[",
    kegg>=100 & kegg < 300 ~ "[100-300[",
    kegg>=300   ~ ">300"
  ) ) %>%
  mutate(kegg_break =  as.factor(kegg_break)) %>%
  mutate(kegg_break =  forcats::fct_relevel(kegg_break, "[100-300[", after=3 )) %>%
  ggplot() + 
  geom_bar(aes(x=kegg_break,fill=dist_to_lr10%>%as.character), position="fill") + 
  scale_fill_viridis_d("MSP edge distance\nto Test product species",option="magma") + 
  xlab("Number of KOs in common with\nTest product species contributions") +
  ylab("MSP proportion") + theme_minimal()

kegg_function_plot


```


```{r fig.height=14, fig.width=16}
cowplot::plot_grid(kegg_modules_plot, kegg_brite_barplot, network_plot,kegg_function_plot, nrow=3, rel_heights = c(1.2,2), rel_widths = c(1.5,1), labels = "AUTO")

ggsave("Figure_MSP_network_KEGG_contribution.pdf")

```



