---
title: "Richness analysis - metagenomic shotgun data"
output: html_notebook
---

```{r message=FALSE, warning=FALSE}

library(dplyr)
library(DESeq2)
library(ggplot2)
library(foreach)
library(readr)
library(ggpubr)
library(ggsignif)
library(reshape)
devtools::load_all(reset=FALSE)
library(cowplot)
library(grid)

```

# gene count 
```{r}

load("~/tolerance/data-raw/profile_tolerance_scaled_rounded_dwzn13.2M_final.RData")

gene_count_tolerance_dwzn13.2M = colSums(profile_tolerance_scaled_rounded_dwzn13.2M > 0)
save(gene_count_tolerance_dwzn13.2M, file="gene_count_tolerance_13.2M.rda")

Tolerance::sample_metadata_metaG  %>% head
sample_metadata_metaG$gene_count_dwzn13.2M = gene_count_tolerance_dwzn13.2M
save(sample_metadata_metaG, file="~/tolerance/data/sample_metadata_metaG.rda")

load("~/tolerance/data-raw/profile_tolerance_scaled_rounded_dwzn16M_final.RData")

gene_count_tolerance_dwzn16M = colSums(profile_tolerance_scaled_rounded_dwzn16M > 0)
save(gene_count_tolerance_dwzn16M, file="gene_count_tolerance_dwzn16M.rda")

Tolerance::sample_metadata_metaG  %>% head
sample_metadata_metaG$gene_count_tolerance_dwzn16M = gene_count_tolerance_dwzn16M
save(sample_metadata_metaG, file="~/tolerance/data/sample_metadata_metaG.rda")


```

# gene richness according to product consumption
```{r}
# with all samples

tapply(sample_metadata_metaG$gene_count_dwzn13.2M,sample_metadata_metaG$group, summary)


df_sample_metadata_metaG = data.frame(sample_metadata_metaG[,c("samplenr","gene_count_dwzn13.2M", "group", "Visit", "Groupe_dose")])


m.df_sample_metadata_metaG = melt(df_sample_metadata_metaG)

my_comparisons <- list( c("V2_ATN.3", "V2_CP.3"), c("V4_ATN.3","V4_CP.3") )

p1 = 
  ggboxplot(m.df_sample_metadata_metaG, x = "Groupe_dose", y = "value",
              color = "Groupe_dose", 
            fill="Groupe_dose",
            alpha=0.5,
              palette=c("#66CCFF", "#3399FF", "#FF9999", "#FF0033"),
              add = "jitter",
              ylab = "Gene richness",
              xlab= "",
              ggtheme = theme_pubr()) +
# scale_y_continuous(name = 'Gene richness', limits=c(0,1200000)) +
   stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
   theme(axis.title.x=element_blank(), legend.position = 'none')+
  facet_grid(. ~ Visit)+
  theme(strip.text.x = element_text(size=13, color="white", face="bold")) +
  theme(strip.background = element_rect(colour="white", fill="gray48", linetype="solid"))+
  theme(axis.title.x = element_text(color="black", face="bold", size=13),
        axis.title.y = element_text(color="black", size=12, face="bold"),
        axis.text.y = element_text(color="black", size=10, face="bold"),
        axis.text.x = element_text(color="black", size=10, face="bold"),
        legend.text = element_text(color="black", size=10, face="bold"),
        legend.title = element_text(color="black", size=13, face="bold"))

```

# metadata file samples with the 2 timepoints v2,v4, no ATN3 V5 
```{r}

load("~/tolerance/data/sample_metadata_metaG_allvisits_v2v4_090719.rda")

load("~/tolerance/data-raw/profile_tolerance_scaled_rounded_dwzn13.2M_final.RData")

profile = profile_tolerance_scaled_rounded_dwzn13.2M
profile = as.data.frame(profile)
profile_v2v4 = profile[,sample_metadata_metaG_allvisits_v2v4$samplenr]


gene_count_tolerance_dwzn13.2M_v2v4 = colSums(profile_v2v4 > 0)
save(gene_count_tolerance_dwzn13.2M_v2v4, file="~/tolerance/data/gene_count_tolerance_dwzn13.2M_v2v4.rda")

Tolerance::sample_metadata_metaG_allvisits_v2v4  %>% head
sample_metadata_metaG_allvisits_v2v4$gene_count = gene_count_tolerance_dwzn13.2M_v2v4
save(sample_metadata_metaG_allvisits_v2v4, file="~/tolerance/data/sample_metadata_metaG_allvisits_v2v4.rda")

load("~/tolerance/data/sample_metadata_metaG_allvisits_v2v4.rda")

sample_metadata_metaG_allvisits_v2v4$group = as.factor(sample_metadata_metaG_allvisits_v2v4$group)
sample_metadata_metaG_allvisits_v2v4$Visit = as.factor(sample_metadata_metaG_allvisits_v2v4$Visit)


df_sample_metadata_metaG_allvisits_v2v4 = data.frame(sample_metadata_metaG_allvisits_v2v4[,c("samplenr","gene_count", "group", "Visit", "Groupe_dose")])


m.df_sample_metadata_metaG_allvisits_v2v4 = melt(df_sample_metadata_metaG_allvisits_v2v4)


m.df_sample_metadata_metaG_allvisits_v2v4_filterv2 = m.df_sample_metadata_metaG_allvisits_v2v4 %>% filter(Visit == "V2")
m.df_sample_metadata_metaG_allvisits_v2v4_filterv2 = transform(m.df_sample_metadata_metaG_allvisits_v2v4_filterv2, Visit= factor(Visit, labels="D0"))
  
m.df_sample_metadata_metaG_allvisits_v2v4_filterv2 = transform(m.df_sample_metadata_metaG_allvisits_v2v4_filterv2, Groupe_dose= factor(Groupe_dose, labels=c("CP.3","FMP.3"), ordered=TRUE))
  
  
my_comparisons_v2 <- list( c("FMP.3", "CP.3"))

p1=
 ggboxplot(m.df_sample_metadata_metaG_allvisits_v2v4_filterv2, x = "Groupe_dose", y = "value",
              color = "Groupe_dose", 
            fill="Groupe_dose",
            alpha=0.5,
              # palette=c("#66CCFF", "#3399FF", "#FF9999", "#FF0033"),
           palette=c("#66CCFF", "#FF9999"),
              add = "jitter",
              ylab = "Gene richness",
              xlab= "",
              ggtheme = theme_pubr()) +
scale_y_continuous(name = 'Gene richness', limits=c(0,1300000)) +
   stat_compare_means(comparisons = my_comparisons_v2, label = "p.signif")+
   theme(axis.title.x=element_blank(), legend.position = 'none')+
 facet_grid(. ~ Visit)+
  theme(strip.text.x = element_text(size=13, color="white", face="bold")) +
  theme(strip.background = element_rect(colour="white", fill="gray48", linetype="solid"))+
  theme(axis.title.x = element_text(color="black", face="bold", size=13),
        axis.title.y = element_text(color="black", size=12, face="bold"),
        axis.text.y = element_text(color="black", size=10, face="bold"),
        axis.text.x = element_text(color="black", size=10, face="bold"),
        legend.text = element_text(color="black", size=10, face="bold"),
        legend.title = element_text(color="black", size=13, face="bold"))


m.df_sample_metadata_metaG_allvisits_v2v4_filterv4 = m.df_sample_metadata_metaG_allvisits_v2v4 %>% filter(Visit == "V4")
m.df_sample_metadata_metaG_allvisits_v2v4_filterv4 = transform(m.df_sample_metadata_metaG_allvisits_v2v4_filterv4, Visit= factor(Visit, labels="D28"))
  
m.df_sample_metadata_metaG_allvisits_v2v4_filterv4 = transform(m.df_sample_metadata_metaG_allvisits_v2v4_filterv4, Groupe_dose= factor(Groupe_dose, labels=c("CP.3","FMP.3"), ordered=TRUE))
  
  
my_comparisons_v4 <- list( c("FMP.3", "CP.3"))


p2=
 ggboxplot(m.df_sample_metadata_metaG_allvisits_v2v4_filterv4, x = "Groupe_dose", y = "value",
              color = "Groupe_dose", 
            fill="Groupe_dose",
            alpha=0.5,
              # palette=c("#66CCFF", "#3399FF", "#FF9999", "#FF0033"),
           palette=c("#3399FF", "#FF0033"),
              add = "jitter",
              ylab = "Gene richness",
              xlab= "",
              ggtheme = theme_pubr()) +
scale_y_continuous(name = 'Gene richness', limits=c(0,1300000)) +
   stat_compare_means(comparisons = my_comparisons_v4, label = "p.signif")+
   theme(axis.title.x=element_blank(), legend.position = 'none')+
 facet_grid(. ~ Visit)+
  theme(strip.text.x = element_text(size=13, color="white", face="bold")) +
  theme(strip.background = element_rect(colour="white", fill="gray48", linetype="solid"))+
  theme(axis.title.x = element_text(color="black", face="bold", size=13),
        axis.title.y = element_text(color="black", size=12, face="bold"),
        axis.text.y = element_text(color="black", size=10, face="bold"),
        axis.text.x = element_text(color="black", size=10, face="bold"),
        legend.text = element_text(color="black", size=10, face="bold"),
        legend.title = element_text(color="black", size=13, face="bold"))

tiff(file="~/tolerance/notebook/gene_richness_samplesv2v4.tiff", w=800, h=400)
plot_grid(p1,p2,align="h",nrow=1, ncol=2)
dev.off()



# affichage stats

  
my_comparisons_v2 <- list( c("ATN.3", "CP.3"))

p1=
 ggboxplot(m.df_sample_metadata_metaG_allvisits_v2v4_filterv2, x = "Groupe_dose", y = "value",
              color = "Groupe_dose", 
            fill="Groupe_dose",
            alpha=0.5,
              # palette=c("#66CCFF", "#3399FF", "#FF9999", "#FF0033"),
           palette=c("#66CCFF", "#FF9999"),
              add = "jitter",
              ylab = "Gene richness",
              xlab= "",
              ggtheme = theme_pubr()) +
scale_y_continuous(name = 'Gene richness', limits=c(0,1300000)) +
   # stat_compare_means(comparisons = my_comparisons_v2, method = "wilcox.test", paired = F)+
  stat_compare_means()+
   theme(axis.title.x=element_blank(), legend.position = 'none')+
 facet_grid(. ~ Visit)+
  theme(strip.text.x = element_text(size=13, color="white", face="bold")) +
  theme(strip.background = element_rect(colour="white", fill="gray48", linetype="solid"))+
  theme(axis.title.x = element_text(color="black", face="bold", size=13),
        axis.title.y = element_text(color="black", size=12, face="bold"),
        axis.text.y = element_text(color="black", size=10, face="bold"),
        axis.text.x = element_text(color="black", size=10, face="bold"),
        legend.text = element_text(color="black", size=10, face="bold"),
        legend.title = element_text(color="black", size=13, face="bold"))



my_comparisons_v4 <- list( c("ATN.3", "CP.3"))

p2=
 ggboxplot(m.df_sample_metadata_metaG_allvisits_v2v4_filterv4, x = "Groupe_dose", y = "value",
              color = "Groupe_dose", 
            fill="Groupe_dose",
            alpha=0.5,
              # palette=c("#66CCFF", "#3399FF", "#FF9999", "#FF0033"),
           palette=c("#3399FF", "#FF0033"),
              add = "jitter",
              ylab = "Gene richness",
              xlab= "",
              ggtheme = theme_pubr()) +
scale_y_continuous(name = 'Gene richness', limits=c(0,1300000)) +
   stat_compare_means()+
   theme(axis.title.x=element_blank(), legend.position = 'none')+
 facet_grid(. ~ Visit)+
  theme(strip.text.x = element_text(size=13, color="white", face="bold")) +
  theme(strip.background = element_rect(colour="white", fill="gray48", linetype="solid"))+
  theme(axis.title.x = element_text(color="black", face="bold", size=13),
        axis.title.y = element_text(color="black", size=12, face="bold"),
        axis.text.y = element_text(color="black", size=10, face="bold"),
        axis.text.x = element_text(color="black", size=10, face="bold"),
        legend.text = element_text(color="black", size=10, face="bold"),
        legend.title = element_text(color="black", size=13, face="bold"))

tiff(file="~/tolerance/notebook/gene_richness_samplesv2v4_stats.tiff", w=800, h=400)
plot_grid(p1,p2,align="h",nrow=1, ncol=2)
dev.off()


# fonction lineaire suggerer par marion poirel pour analyser l'alpha diversite (gene richness)

lm_gene_richness= lm(m.df_sample_metadata_metaG_allvisits_v2v4_filterv4$value~m.df_sample_metadata_metaG_allvisits_v2v4_filterv2$value+m.df_sample_metadata_metaG_allvisits_v2v4_filterv4$Groupe_dose)

summary(lm_gene_richness)

Residuals:
    Min      1Q  Median      3Q     Max 
-447313  -58339   14343   90513  184040 

Coefficients:
                                                                      Estimate
(Intercept)                                                          3.413e+05
m.df_sample_metadata_metaG_allvisits_v2v4_filterv2$value             5.263e-01
m.df_sample_metadata_metaG_allvisits_v2v4_filterv4$Groupe_doseATN.3 -7.821e+03
                                                                    Std. Error t value
(Intercept)                                                          1.110e+05   3.074
m.df_sample_metadata_metaG_allvisits_v2v4_filterv2$value             1.284e-01   4.098
m.df_sample_metadata_metaG_allvisits_v2v4_filterv4$Groupe_doseATN.3  4.536e+04  -0.172
                                                                    Pr(>|t|)    
(Intercept)                                                         0.004075 ** 
m.df_sample_metadata_metaG_allvisits_v2v4_filterv2$value            0.000235 ***
m.df_sample_metadata_metaG_allvisits_v2v4_filterv4$Groupe_doseATN.3 0.864105    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 138700 on 35 degrees of freedom
Multiple R-squared:  0.3296,	Adjusted R-squared:  0.2913 
F-statistic: 8.602 on 2 and 35 DF,  p-value: 0.0009147



## je ne comprends pas


```

# MSP count

```{r}

# all samples (107 samples)
load("~/tolerance/data/MSP_count_matrix_rounded.rda")
mgs_count = colSums(MSP_count_matrix_rounded > 0)
save(mgs_count, file="~/tolerance/data/mgs_count.rda")
write.csv2(mgs_count, file="~/tolerance/data/mgs_count.csv")

Tolerance::sample_metadata_metaG  %>% head
sample_metadata_metaG$mgs_count = colSums(MSP_count_matrix_rounded > 0)
save(sample_metadata_metaG, file="~/tolerance/data/sample_metadata_metaG.rda")

# samples with all visits v2/v4 (76 samples)
load("~/tolerance/data/sample_metadata_metaG_allvisits_v2v4.rda")
Tolerance::sample_metadata_metaG_allvisits_v2v4  %>% head
MSP_count_matrix_rounded_v2v4 = MSP_count_matrix_rounded[,sample_metadata_metaG_allvisits_v2v4$samplenr]
save(MSP_count_matrix_rounded_v2v4, file="~/tolerance/data/MSP_count_matrix_rounded_v2v4.rda")

MSP_count_allvisits_v2v4 = colSums(MSP_count_matrix_rounded_v2v4 > 0)
save(MSP_count_allvisits_v2v4, file="~/tolerance/data/MSP_count_allvisits_v2v4.rda")

Tolerance::sample_metadata_metaG_allvisits_v2v4  %>% head
sample_metadata_metaG_allvisits_v2v4$MSP_count = MSP_count_allvisits_v2v4
save(sample_metadata_metaG_allvisits_v2v4, file="~/tolerance/data/sample_metadata_metaG_allvisits_v2v4.rda")

sample_metadata_metaG_allvisits_v2v4$group = as.factor(sample_metadata_metaG_allvisits_v2v4$group)
sample_metadata_metaG_allvisits_v2v4$Visit = as.factor(sample_metadata_metaG_allvisits_v2v4$Visit)


df_sample_metadata_metaG_allvisits_v2v4 = data.frame(sample_metadata_metaG_allvisits_v2v4[,c("samplenr","MSP_count", "group", "Visit", "Groupe_dose")])

m.df_sample_metadata_metaG_allvisits_v2v4 = melt(df_sample_metadata_metaG_allvisits_v2v4)

m.df_sample_metadata_metaG_allvisits_v2v4_filterv2 = m.df_sample_metadata_metaG_allvisits_v2v4 %>% filter(Visit == "V2")
m.df_sample_metadata_metaG_allvisits_v2v4_filterv2 = transform(m.df_sample_metadata_metaG_allvisits_v2v4_filterv2, Visit= factor(Visit, labels="D0"))
  
m.df_sample_metadata_metaG_allvisits_v2v4_filterv2 = transform(m.df_sample_metadata_metaG_allvisits_v2v4_filterv2, Groupe_dose= factor(Groupe_dose, labels=c("CP.3","FMP.3"), ordered=TRUE))
  
  
my_comparisons_v2 <- list( c("FMP.3", "CP.3"))

p3=
 ggboxplot(m.df_sample_metadata_metaG_allvisits_v2v4_filterv2, x = "Groupe_dose", y = "value",
              color = "Groupe_dose", 
            fill="Groupe_dose",
            alpha=0.5,
              # palette=c("#66CCFF", "#3399FF", "#FF9999", "#FF0033"),
           palette=c("#66CCFF", "#FF9999"),
              add = "jitter",
              ylab = "MSP richness",
              xlab= "",
              ggtheme = theme_pubr()) +
scale_y_continuous(name = 'MSP richness', limits=c(0,300)) +
   stat_compare_means(comparisons = my_comparisons_v2, label = "p.signif")+
   theme(axis.title.x=element_blank(), legend.position = 'none')+
 facet_grid(. ~ Visit)+
  theme(strip.text.x = element_text(size=13, color="white", face="bold")) +
  theme(strip.background = element_rect(colour="white", fill="gray48", linetype="solid"))+
  theme(axis.title.x = element_text(color="black", face="bold", size=13),
        axis.title.y = element_text(color="black", size=12, face="bold"),
        axis.text.y = element_text(color="black", size=10, face="bold"),
        axis.text.x = element_text(color="black", size=10, face="bold"),
        legend.text = element_text(color="black", size=10, face="bold"),
        legend.title = element_text(color="black", size=13, face="bold"))


m.df_sample_metadata_metaG_allvisits_v2v4_filterv4 = m.df_sample_metadata_metaG_allvisits_v2v4 %>% filter(Visit == "V4")
m.df_sample_metadata_metaG_allvisits_v2v4_filterv4 = transform(m.df_sample_metadata_metaG_allvisits_v2v4_filterv4, Visit= factor(Visit, labels="D28"))

m.df_sample_metadata_metaG_allvisits_v2v4_filterv4 = transform(m.df_sample_metadata_metaG_allvisits_v2v4_filterv4, Groupe_dose= factor(Groupe_dose, labels=c("CP.3","FMP.3"), ordered=TRUE))
  
  
my_comparisons_v4 <- list( c("FMP.3", "CP.3"))

p4=
 ggboxplot(m.df_sample_metadata_metaG_allvisits_v2v4_filterv4, x = "Groupe_dose", y = "value",
              color = "Groupe_dose", 
            fill="Groupe_dose",
            alpha=0.5,
              # palette=c("#66CCFF", "#3399FF", "#FF9999", "#FF0033"),
           palette=c("#3399FF", "#FF0033"),
              add = "jitter",
              ylab = "MSP richness",
              xlab= "",
              ggtheme = theme_pubr()) +
scale_y_continuous(name = 'MSP richness', limits=c(0,300)) +
   stat_compare_means(comparisons = my_comparisons_v4, label = "p.signif")+
   theme(axis.title.x=element_blank(), legend.position = 'none')+
 facet_grid(. ~ Visit)+
  theme(strip.text.x = element_text(size=13, color="white", face="bold")) +
  theme(strip.background = element_rect(colour="white", fill="gray48", linetype="solid"))+
  theme(axis.title.x = element_text(color="black", face="bold", size=13),
        axis.title.y = element_text(color="black", size=12, face="bold"),
        axis.text.y = element_text(color="black", size=10, face="bold"),
        axis.text.x = element_text(color="black", size=10, face="bold"),
        legend.text = element_text(color="black", size=10, face="bold"),
        legend.title = element_text(color="black", size=13, face="bold"))

tiff(file="~/tolerance/notebook/MSP_richness_samplesv2v4.tiff", w=800, h=400)
plot_grid(p3,p4,align="h",nrow=1, ncol=2)
dev.off()

# combine gene and MSP richness on the same plot



tiff(file="~/tolerance/notebook/microbiome_richness_samplesv2v4.tiff", w=900, h=700)
plot_grid(p1,p2,p3,p4,align="h",labels = c("A","", "B", ""),nrow=2, ncol=2)
dev.off()

```


```{r}
# delta gene count
# delta msp count
load("~/tolerance/data/sample_metadata_metaG_allvisits_v2v4.rda")
write.csv2(sample_metadata_metaG_allvisits_v2v4, file="~/tolerance/data/sample_metadata_metaG_allvisits_v2v4.csv")

# delta gene count and MSP count on excel

sample_metadata_metaG_allvisits_v2v4_delta = read.csv2("~/tolerance/data/sample_metadata_metaG_allvisits_v2v4_deltaGC_MSP.csv")



df_sample_metadata_metaG_allvisits_v2v4_delta = data.frame(sample_metadata_metaG_allvisits_v2v4_delta[,c("delta_gene_count", "group", "Visit", "Groupe_dose")])

m.df_sample_metadata_metaG_allvisits_v2v4_delta = melt(df_sample_metadata_metaG_allvisits_v2v4_delta)

m.df_sample_metadata_metaG_allvisits_v2v4_deltafilterv2 = m.df_sample_metadata_metaG_allvisits_v2v4_delta %>% filter(Visit == "V2")


# m.df_sample_metadata_metaG_allvisits_v2v4_delta = transform(m.df_sample_metadata_metaG_allvisits_v2v4_delta, Visit= factor(Visit, labels=c("D0","D28"), order=T))
  
my_comparisons_deltagene <- list( c("ATN.3", "CP.3"))

p5=
 ggboxplot(m.df_sample_metadata_metaG_allvisits_v2v4_deltafilterv2, x = "Groupe_dose", y = "value",
              color = "Groupe_dose", 
            fill="Groupe_dose",
            alpha=0.5,
              # palette=c("#66CCFF", "#3399FF", "#FF9999", "#FF0033"),
           palette=c("#3399FF", "#FF0033"),
              add = "jitter",
              ylab = "Delta gene richness",
              xlab= "",
              ggtheme = theme_pubr()) +
# scale_y_continuous(name = 'Delta gene richness', limits=c(0,300)) +
   stat_compare_means(comparisons = my_comparisons_deltagene, label = "p.signif")+
   theme(axis.title.x=element_blank(), legend.position = 'none')+
 # facet_grid(. ~ Visit)+
  theme(strip.text.x = element_text(size=13, color="white", face="bold")) +
  theme(strip.background = element_rect(colour="white", fill="gray48", linetype="solid"))+
  theme(axis.title.x = element_text(color="black", face="bold", size=13),
        axis.title.y = element_text(color="black", size=12, face="bold"),
        axis.text.y = element_text(color="black", size=10, face="bold"),
        axis.text.x = element_text(color="black", size=10, face="bold"),
        legend.text = element_text(color="black", size=10, face="bold"),
        legend.title = element_text(color="black", size=13, face="bold"))



df_sample_metadata_metaG_allvisits_v2v4_delta = data.frame(sample_metadata_metaG_allvisits_v2v4_delta[,c("delta_MSP_count", "group", "Visit", "Groupe_dose")])

m.df_sample_metadata_metaG_allvisits_v2v4_delta = melt(df_sample_metadata_metaG_allvisits_v2v4_delta)

m.df_sample_metadata_metaG_allvisits_v2v4_deltafilterv2 = m.df_sample_metadata_metaG_allvisits_v2v4_delta %>% filter(Visit == "V2")

# m.df_sample_metadata_metaG_allvisits_v2v4_delta = transform(m.df_sample_metadata_metaG_allvisits_v2v4_delta, Visit= factor(Visit, labels=c("D0","D28"), order=T))
#   
my_comparisons_deltamsp <- list( c("ATN.3", "CP.3"))

p6=
 ggboxplot(m.df_sample_metadata_metaG_allvisits_v2v4_deltafilterv2, x = "Groupe_dose", y = "value",
              color = "Groupe_dose", 
            fill="Groupe_dose",
            alpha=0.5,
              # palette=c("#66CCFF", "#3399FF", "#FF9999", "#FF0033"),
           palette=c("#3399FF", "#FF0033"),
              add = "jitter",
              ylab = "Delta MSP richness",
              xlab= "",
              ggtheme = theme_pubr()) +
# scale_y_continuous(name = 'Delta gene richness', limits=c(0,300)) +
   stat_compare_means(comparisons = my_comparisons_deltamsp, label = "p.signif")+
   theme(axis.title.x=element_blank(), legend.position = 'none')+
 # facet_grid(. ~ Visit)+
  theme(strip.text.x = element_text(size=13, color="white", face="bold")) +
  theme(strip.background = element_rect(colour="white", fill="gray48", linetype="solid"))+
  theme(axis.title.x = element_text(color="black", face="bold", size=13),
        axis.title.y = element_text(color="black", size=12, face="bold"),
        axis.text.y = element_text(color="black", size=10, face="bold"),
        axis.text.x = element_text(color="black", size=10, face="bold"),
        legend.text = element_text(color="black", size=10, face="bold"),
        legend.title = element_text(color="black", size=13, face="bold"))



tiff(file="~/tolerance/notebook/delta_microbiome_richness_samplesv2v4.tiff", w=800, h=400)
plot_grid(p5,p6,align="h",labels = c("A", "B"),nrow=1, ncol=2)
dev.off()


```
